src/app/core/models/system-user.model.ts

```diff
@@ -1,7 +1,10 @@
+import { RoleModel } from "./role.model";
+
 export interface SystemUserModel {
```

Comment:
Você não deve ter uma lista de roles dentro desse model. Você deverá criar um outro model para ser usado quando precisar dessa lista de roles. Para isso faça algo similar a isso em outro arquivo.

```suggestion
export interface SystemUserSystemRoleModel extends SystemUserModel {
      roles:RoleModel[];
}
```